go/processor/floria/processor.go

```diff
@@ -91,6 +101,10 @@ func (p *processor) Run(
 	callParameters := callParameters(transaction, gas)
 	kind := callKind(transaction)
 
+	if kind == tosca.Call {
```

Comment:
This is the processor run function, it is called once per transaction in the beginning, recursive calls do not end up here. At this point the transaction is executed therefore the state nonce has to be increased. 
It can not be increased earlier as the transaction could fail in due to a check. It is incremented inside the call for create instructions as there is a value transfer check which could terminate the transaction early.